core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java

```diff
@@ -149,6 +166,65 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static MultiGetResponse fromXContent(XContentParser parser) throws IOException {
+        String currentFieldName = null;
+        List<MultiGetItemResponse> items = new ArrayList<>();
+        for (Token token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
+            switch (token) {
+                case FIELD_NAME:
+                    currentFieldName = parser.currentName();
+                    break;
+                case START_ARRAY:
+                    if (Fields.DOCS.equals(currentFieldName)) {
+                        for (token = parser.nextToken(); token != Token.END_ARRAY; token = parser.nextToken()) {
+                            if (token == Token.START_OBJECT) {
+                                String index = null;
+                                String type = null;
+                                String id = null;
+                                ElasticsearchException exception = null;
+                                GetResult getResult = null;
+                                item: for (token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
```

Comment:
Interesting, haven't seen breaks with targets in a while. This might be a bit too sneaky, I think I would like pulling out the item parser for better readability.